sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -127,9 +133,17 @@ object GpuDeviceManager extends Logging {
   }
 
   def shutdown(): Unit = synchronized {
-    RapidsBufferCatalog.close()
-    Rmm.shutdown()
-    singletonMemoryInitialized = false
+    try {
+      RapidsBufferCatalog.close()
+      Rmm.shutdown()
+      singletonMemoryInitialized = Uninitialized
+    } catch {
+      case e: Throwable =>
+        // flag as Uninitializable so that any future initialization attempts can throw
+        // a helpful error message
+        singletonMemoryInitialized = Uninitializable
+        throw e
+    }
```

Comment:
Can simplify this by assuming shutdown will fail up front
```suggestion
    // assume error during shutdown until we complete it
    singletonMemoryInitialied = Uninitializable
    RapidsBufferCatalog.close()
    Rmm.shutdown()
    singletonMemoryInitialized = Uninitialized
```